src/compile/mod.rs

```diff
@@ -174,15 +175,15 @@ impl CompiledProgram {
         int_offset: usize,
         ext_offset: usize,
         func_offset: usize,
-        globals_offset: usize,
+        globals_offset: Vec<GlobalVarDecl>,
```

Comment:
The variable name `globals_offset` is confusing here. Maybe change it to `globals`?